### PR TITLE
Minor improvements and gitignore added

### PR DESCRIPTION
**Pull request by PanderOpenTaal**

### CODE REVIEW

**Comment by PanderOpenTaal** [comment]
Note also the following warning four times in the output of make regarding the use of %lu instead of %u in printf.

```
ar: `u' modifier ignored since `D' is the default (see `U')
```

File: src/enchant.h [hunk 1]
```diff
@@ -52,7 +52,7 @@ typedef struct str_enchant_broker EnchantBroker;
 typedef struct str_enchant_dict   EnchantDict;
 
 /* const */
-ENCHANT_MODULE_EXPORT (char *)
+ENCHANT_MODULE_EXPORT (const char *)
```
Comment (TingPing) [inline src/enchant.h]: API break?

Comment (hfiguiere) [inline src/enchant.h]: One thing I'm sure is that it was invalid. gcc warnings should already have complained.

**Comment by PanderMusubi** [comment]
Thanks for responding so quickly.

Meanwhile, I have reverted that particular change and will provide a new pull request shortly. It worked for me, otherwise gcc gives this warning:

```
enchant.c: In function 'enchant_get_version':
<command-line>:0:24: warning: return discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
enchant.c:2338:9: note: in expansion of macro 'ENCHANT_VERSION_STRING'
  return ENCHANT_VERSION_STRING;
         ^
```

Any other way how to get rid of this warning?

Second question, is the following. The tests report this:

```
$ ./enchant
Usage: /home/pander/workspace/enchant/tests/.libs/lt-enchant [options] -a|-d dict|-l|-L|-m|-v[v]|<file>
    -a lists alternatives.
    -d dict uses dictionary <dict>.
    -l lists misspellings.
    -m is ignored.
    -L displays line numbers.
    -v displays program version.
```

Which looks fine.

```
$ ./test-enchant
Couldn't create dictionary for en_US
Personal Wordlist: Personal Wordlist 'Personal Wordlist' (test.pwl)
enchant_dict_check (hello): found
enchant_dict_check (helllo): found
enchant_dict_suggest(helllo): 1
    =>helllo
enchant_dict_suggest(taag): 1
    =>tag
Adding 'helllo' to session
enchant_dict_check (hello): found
enchant_dict_check (helllo): found
Adding 'helllo' to personal
enchant_dict_check (hello): found
enchant_dict_check (helllo): found

$ ./test-enchantxx 
Couldn't create dictionary for en_US: 
```

How can I get rid of the error regarding en_US for the two previous examples? I have such dictionary installed, see

```
$ dpkg -l|grep -i spell|grep -i us
ii  hunspell-en-us                              20070829-6ubuntu2                          all          English_american dictionary for hunspell
```

**Comment by TingPing** [comment]
> Any other way how to get rid of this warning?

The problem is if you fix the API you move the warning (or even error) to users of the library who didn't use const. So you would just have to ignore it I'd think, however wrong that might be.

**Comment by hfiguiere** [comment]
But then they could fix it. It is not more wrong on this side of the code.... and they might have to fix bugs.

**Comment by PanderMusubi** [comment]
Would be good to fix it one day so that all users of the API maintain their code.

Any pointers on how to fix the error in my tests on creating dictionary en_US?

**Comment by PanderOpenTaal** [comment]
Please, review the latest push on my master branch for this pull request.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+libtool
+ltmain.sh
+missing
+Makefile
+Makefile.in
+src/**/*.la
+src/**/*.lo
+src/**/*.o
+src/**/.libs
+src/**/.deps
```
Comment (hfiguiere) [inline .gitignore:24]: Since these are generic can just remove the path and do

```
*.la
*.lo
*.o
.libs
.deps
```

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+src/libenchant.rc
+tests/**/*.la
+tests/**/*.lo
+tests/**/*.o
+tests/**/.libs
+tests/**/.deps
```
Comment (hfiguiere) [inline .gitignore:30]: and then remove these.

File: src/pwl.c [hunk 4]
```diff
@@ -322,7 +322,7 @@ static void enchant_pwl_refresh_from_file(EnchantPWL* pwl)
 				line[l] = '\0';
 			else if(!feof(f)) /* ignore lines longer than BUFSIZ. */ 
 				{
-					g_warning ("Line too long (ignored) in %s at line:%u\n", pwl->filename, line_number);
+					g_warning ("Line too long (ignored) in %s at line:%lu\n", pwl->filename, line_number);
```
Comment (hfiguiere) [inline src/pwl.c:325]: `line_number` is a `size_t` so what you want here is %zu.

File: src/pwl.c [hunk 5]
```diff
@@ -336,7 +336,7 @@ static void enchant_pwl_refresh_from_file(EnchantPWL* pwl)
 					if(g_utf8_validate(line, -1, NULL))
 						enchant_pwl_add_to_trie(pwl, line, strlen(line));
 					else
-						g_warning ("Bad UTF-8 sequence in %s at line:%u\n", pwl->filename, line_number);
+						g_warning ("Bad UTF-8 sequence in %s at line:%lu\n", pwl->filename, line_number);
```
Comment (hfiguiere) [inline src/pwl.c:339]: and here

File: tests/test-enchant.c [hunk 6]
```diff
@@ -81,7 +81,7 @@ run_dict_tests (EnchantDict * dict)
 				enchant_dict_suggest (dict, sugg_checks[i], strlen (sugg_checks[i]),
 						      &n_suggs);
 			
-			printf ("enchant_dict_suggest(%s): %d\n", sugg_checks[i], n_suggs);
+			printf ("enchant_dict_suggest(%s): %lu\n", sugg_checks[i], n_suggs);
```
Comment (hfiguiere) [inline tests/test-enchant.c:84]: and here

File: tests/test-enchantxx.cpp [hunk 7]
```diff
@@ -69,7 +69,7 @@ run_dict_tests (enchant::Dict * dict)
 		{
 			dict->suggest (sugg_checks[i], suggs);
 			
-			printf ("enchant_dict_suggest(%s): %d\n", sugg_checks[i], suggs.size());
+			printf ("enchant_dict_suggest(%s): %lu\n", sugg_checks[i], suggs.size());
```
Comment (hfiguiere) [inline tests/test-enchantxx.cpp:72]: also here, %zu.

**Comment by hfiguiere** [comment]
Sorry for the late reply. Please see the comments.

Also, since you have a commit to reverse stuff, I'd rather see it be gone and amend that actual commit. You can push force your own branch.

Thanks for contributing.

**Comment by hfiguiere** [comment]
PR #10 got the .gitignore merged

**Comment by rrthomas** [comment]
I believe this pull request can now be closed, as I've addressed all the issues in it with other pull requests. Thanks, @PanderOpenTaal!

**Pull request closed without merging**